src/main/java/reposense/RepoSense.java

```diff
@@ -84,12 +84,8 @@ public static void main(String[] args) {
      */
     public static List<RepoConfiguration> getRepoConfigurations(LocationsCliArguments cliArguments) {
         List<RepoConfiguration> configs = new ArrayList<>();
-        for (String location : cliArguments.getLocations()) {
-            try {
-                configs.add(new RepoConfiguration(location));
-            } catch (InvalidLocationException ile) {
-                logger.log(Level.WARNING, ile.getMessage(), ile);
-            }
+        for (RepoLocation location : cliArguments.getLocations()) {
+            configs.add(new RepoConfiguration(location));
```

Comment:
Sorry, just noticed this from the changed test earlier, the behaviour of our program should not be changing here, from filtering out only invalid 'repo locations', to completely abandoning the process if any1 of the 'repo locations' is invalid.

So in `ArgsParser` and `LocationCliArguments`, we should make them maintain a list of `String`, and we do the conversion to `RepoLocation` here instead before passing it into `RepoConfiguration` (or we can do the conversion within `RepoConfiguration` constructor itself).

@yong24s What you think of this?

Reply:
> So in ArgsParser and LocationCliArguments, we should make them maintain a list of String, and we do the conversion to RepoLocation here instead before passing it into RepoConfiguration (or we can do the conversion within RepoConfiguration constructor itself).

Yes, this implementation is direct and simple.

In the future, we should look into using `ConfigurationBuilder`, where we can delay the validation until `build()` is invoked.